File: Data Structures & Algorithms/duplicate-integer/submission-3.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        if (nums.size() == 0)
            return false;
        for(int i = 0; i < nums.size(); i++){
            for(int j = i + 1; j < nums.size(); j++){
                if (nums[i] > nums[j]){
                    int a = nums[i];
                    nums[i] = nums[j];
                    nums[j] = a;
                }
            }
        }

        for(int i = 0; i < nums.size() - 1; i++){
            if(nums[i] == nums[i+1]){
                return true;
            }
        }
        return false;
    }
};

File: Data Structures & Algorithms/sliding-window-maximum/submission-3.py
class Solution:
    def maxSlidingWindow(self, nums: List[int], k: int) -> List[int]:
        n = len(nums)
        nums = [nums[i] * -1 for i in range(n)]
        output, heap = [], []
        for i in range(k):
            heap.append([nums[i], i])
        heapq.heapify(heap)

        for i in range(k, n):
            output.append(heap[0][0])
            while len(heap) and heap[0][1] <= i - k:
                heapq.heappop(heap)
            heapq.heappush(heap, [nums[i], i])
        output.append(heap[0][0])
        output = [output[i] * -1 for i in range(len(output))]
        return output